pkg/daemon/ceph/osd/agent.go

```diff
@@ -218,6 +232,30 @@ func (a *OsdAgent) configureDevices(context *clusterd.Context, devices *DeviceOs
 	return osds, nil
 }
 
+func getDeviceLVPath(context *clusterd.Context, deviceName string) (string, error) {
+	var lvPath string
+	cmd := fmt.Sprintf("get logical volume path for device")
+	output, err := context.Executor.ExecuteCommandWithOutput(false, cmd, "pvdisplay", "-C", "-o", "lvpath", deviceName, "--reportformat", "json")
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve logical volume path. %+v", err)
+	}
+
+	logger.Debugf("logical volume output : %+v", output)
+	var lvPathResult map[string][]map[string][]LVMInfo
+	err = json.Unmarshal([]byte(output), &lvPathResult)
+	if err != nil {
+		return "", fmt.Errorf("failed to unmarshal logical volume path. %+v", err)
+	}
+
+	if len(lvPathResult["report"][0]["pv"]) == 0 || lvPathResult["report"][0]["pv"][0].LVPath == "" {
```

Comment:
Will `lvPathResult["report"][0]["pv"]` always exist in the json results? If not we could crash.